### ISSUE TITLE
Compression for responses in database

### ISSUE THREAD

**Issue by VasilyKaiser** [caido/caido#971]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help us know what to work on
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Description

Use something like this (https://facebook.github.io/zstd/)  on response in database.
This Rust version is used a lot: https://github.com/gyscos/zstd-rs

### Use cases

When navigating to sites like yahoo, you will get many big responses like for article content.
One such response was 556KB, which was reduced to 65KB by applying FB's fast compression "zstd".
Hence, projects would be much smaller in size.

**Comment by Sytten** [comment]
Somewhat related to https://github.com/caido/caido/issues/893

**Comment by VasilyKaiser** [comment]
"Avoid saving duplicates" - if it's like save only changed part.
Then it sounds like custom compression for http traffic and also hashing could be overhead. 😄
 But adding compression like `zstd` would be a good first actionable step to move closer to the objective.

**Comment by Sytten** [comment]
Yeah I like zstd, there is an sqlite extension for it but it is LGPL so I can't use it. It will be done this year for sure, but I am not sure when since I need to re-code an sqlite extension for it.
